### ISSUE TITLE
What signals are, what signals are available, and how to use them is not documented

### ISSUE THREAD

**Issue by balloonpopper** [godot-escoria/escoria-issues#113]
What signals are, what signals are available, and how to use them is not documented/
I assume "signals" are the standard Godot signals, however Escoria script seems to be able to process at least some of them (eg. arrived). From a script perspective they appear to be referenced the name as events (i.e. <name>: ) but I can't find a list of all the signals that Escoria accepts.

**Comment by dploeger** [comment]
I'm not sure about this. I mean, from an Escoria dev perspective it makes sense. But from a gamedev perspective, we actually would rather hide or abstract these.

@StraToN  @BHSDuncan  what do you think?

**Comment by BHSDuncan** [comment]
I'm in agreement, unless there are UI-related ones we need to document, since the user can very much implement their own. Any signals that are entirely internal to the core should be encapsulated as I think we'd prefer people to treat the core as a black box (or at least closed to change/extension in the vast majority of cases). Maybe if we get to a point where we have a version for people who want to fork and do their own "version" of Escoria, but I think the focus should continue to be on the gamedev.

My $0.02.

**Comment by balloonpopper** [comment]
I only came across it because there's an "arrived" signal used in one of the demo level scripts and I couldn't find documentation to say what it did. If non-custom signals are internal use only I'm fine with that, but references in the demo script should be removed.

**Comment by BHSDuncan** [comment]
Can you point to the file? I'd like to look myself as I never noticed.

**Comment by balloonpopper** [comment]
game/rooms/room03/esc/button.esc::arrived
game/rooms/room03/esc/right_platform.esc::arrived

**Comment by BHSDuncan** [comment]
Ah. This is actually a custom verb/action that shares the same names as a signal (you can add custom verbs via the UI implementation, if I'm not mistaken; e.g. "look" or "use", etc.). This action itself is also used internally (check out action manager), though you're right in that it's being used externally and I don't think is explained anywhere.

@dploeger @StraToN if I'm not explaining this right, please do clarify.

**Comment by balloonpopper** [comment]
If that's a custom verb/action then I'm not sure where it's created. I searched all the files in the demo project, and "arrived" is mentioned in only the 2 x level 3 scripts, and core code. There's no mention of it in any other scene related files, so that implies to me that the demo (currently) relies on the signal from core to function rather than a custom verb/action with the same name.

**Comment by BHSDuncan** [comment]
Remember what I said about looking in the action manager: It'll show you how it's used. Personally, I don't know if it's meant to be used or not as I agree it isn't documented properly (read: at all) and it may just be a holdover/legacy code.

One thing that has happened is that there have been instances of signals, actions/verbs, and events sharing names that are sometimes not meant to accomplish the same things. I can understand why you'd think what you're thinking.

**Comment by balloonpopper** [comment]
Sorry, I'd missed your comment about the action manager earlier. I don't know enough about the code to comment on the action manager turning it into something custom - happy to take your word for it 😄 
I wonder if there's anything else like "action" that needs to be documented? I can see other consts for "walk" and "exit_scene" in the action manager. These appear in the documentation as "consts" but there's no immediately obvious explanation around how to use them.

**Comment by BHSDuncan** [comment]
I meant more that the action manager handles a few internal actions, including "arrived". Anything "custom" is handled automatically and is "implemented" by doing so in a custom UI (check out the 9-verbs plugin). I didn't mean quite as much to go through them to determine other actions, etc. :)

"walk" is used by the "walk*" ESC commands.

"exit_scene" is very similar to the case of "arrived".

@dploeger @StraToN I definitely think we need to either add those actions into the ESC command docs, or otherwise explain them (and their usage, if any) in the docs.

**Comment by dploeger** [comment]
Agreed. So this issue should be titled "Document action workflow" or something? I thought we had something like this, but I can't find it.
